### PR TITLE
Add IPv4 validator

### PR DESCRIPTION
**Pull request by parshikov**
Validates that a value is ipv4 address in valid range
```php
use Phalcon\Mvc\Model;
use Phalcon\Mvc\Model\Validator\IPv4;

class Server extends Phalcon\Mvc\Model
{
     public function validation()
     {
          $this->validate(new IPv4(array(
               'field' => 'ipv4address',
               'message' => 'Incorrect ipv4 address',
          )));
 
          if ($this->validationHasFailed() == true) {
               return false;
          }
      }
}

```

### CODE REVIEW

File: Library/Phalcon/Mvc/Model/Validator/IPv4.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        $field = $this->getOption('field');
+
+        if (false === is_string($field)) {
+            throw new Exception('Field name must be a string');
+        }
+
+        $fieldValue = $record->readAttribute($field);
+        $this->getOption('message') ?: 'IPv4 is incorrect';
```
Comment (sergeyklay) [inline Library/Phalcon/Mvc/Model/Validator/IPv4.php]:
```php
$message = $this->getOption('message') ?: 'IPv4 is incorrect';
```

Comment (parshikov) [inline Library/Phalcon/Mvc/Model/Validator/IPv4.php]: Oops

**Comment by endeveit** [comment]
Why did you use regular expression instead of internal [filter_var](http://php.net/filter_var) function with FILTER_VALIDATE_IP flag?

**Comment by parshikov** [comment]
@endeveit You're right. I got this code as a legacy from another project and I've been using it for a while without checking out for efficiency. I'm going to rewrite it soon.

**Pull request closed without merging**